### ISSUE TITLE
Setting of training Kinetics

### ISSUE THREAD

**Issue by vateye** [kenshohara/3D-ResNets-PyTorch#62]
Nice work !
Can you tell me the setting of training Kinetics from scratch? For example learning rate and etc.

**Comment by kenshohara** [comment]
I used the default parameters to train models on Kinetics.

**Comment by yuta1125tp** [comment]
To reproduce the accuracy in your paper, I trained 3D-ResNet18 from scratch on Kinetics dataset. However, my accuracy is low.
As you mentioned in [this issue](https://github.com/kenshohara/3D-ResNets-PyTorch/issues/29#issuecomment-361805646), I changed --n_epochs option from 200(default) to 250.
I also changed --resnet_shortcut option from B(default) to A, as described in [README.md#L79](https://raw.githubusercontent.com/kenshohara/3D-ResNets-PyTorch/master/README.md#L79).
I preprocessed input frames with your scripts(utils/video_jpg_kinetics.py, utils/n_frames_kinetics.py,  utils/kinetics_json.py). I computed my accuracy with your script (utils/eval_kinetics.py).
I use two GTX 1080 Ti.

My opt.json is as follows.
```json
{
    "root_path": "./",
    "video_path": "./data/Kinetics/jpg",
    "annotation_path": "./data/Kinetics/kinetics-400/kinetics_400_activitynet_like.json",
    "result_path": "./works_dlb1x/kinetics400_resnet18",
    "dataset": "kinetics",
    "n_classes": 400,
    "n_finetune_classes": 400,
    "sample_size": 112,
    "sample_duration": 16,
    "initial_scale": 1.0,
    "n_scales": 5,
    "scale_step": 0.84089641525,
    "train_crop": "corner",
    "learning_rate": 0.1,
    "momentum": 0.9,
    "dampening": 0.9,
    "weight_decay": 0.001,
    "mean_dataset": "activitynet",
    "no_mean_norm": false,
    "std_norm": false,
    "nesterov": false,
    "optimizer": "sgd",
    "lr_patience": 10,
    "batch_size": 128,
    "n_epochs": 250,
    "begin_epoch": 1,
    "n_val_samples": 3,
    "resume_path": "",
    "pretrain_path": "",
    "ft_begin_index": 0,
    "no_train": false,
    "no_val": false,
    "test": true,
    "test_subset": "val",
    "scale_in_test": 1.0,
    "crop_position_in_test": "c",
    "no_softmax_in_test": false,
    "no_cuda": false,
    "n_threads": 32,
    "checkpoint": 20,
    "no_hflip": false,
    "norm_value": 1,
    "model": "resnet",
    "model_depth": 18,
    "resnet_shortcut": "A",
    "wide_resnet_k": 2,
    "resnext_cardinality": 32,
    "manual_seed": 1,
    "scales": [
        1.0,
        0.84089641525,
        0.7071067811803005,
        0.5946035574934808,
        0.4999999999911653
    ],
    "arch": "resnet-18",
    "mean": [
        114.7748,
        107.7354,
        99.475
    ],
    "std": [
        38.7568578,
        37.88248729,
        40.02898126
    ]
}
```


Some lines of my train.log is as follows.
```csv
150	2.6882633917304117	0.4012736192511071	0.0010000000000000002
250	2.5532942670860415	0.4291392625660981	1.0000000000000004e-06
```
Some lines of my val.log is as follows.
```csv
150	2.5846365368076936	0.42163036714374613
250	2.537219098577161	0.42939639079029246
```
According to Figure 4.(d) in your paper, my training loss at epoch 150 seems to better than yours. I may have to change some options around learning rate like --lr_patience option to more long. :(


My evaluation results shows as following, and it seems be pretty low.
```python
val_json_file_path = 'works_dlb1x/kinetics400_resnet18/val.json'
gt_json_file_path = 'data/Kinetics/kinetics-400/kinetics_400_activitynet_like.json'

evaluator = eval_kinetics.KINETICSclassification(
    ground_truth_filename=gt_json_file_path,
    prediction_filename=val_json_file_path,
    verbose=True
)
evaluator.evaluate()
>>> [INIT] Loaded annotations from validation subset.
>>> 	Number of ground truth instances: 19906
>>> 	Number of predictions: 133470
>>> [RESULTS] Performance on ActivityNet untrimmed video classification task.
>>> 	Error@1: 0.6321209685521954

evaluator = eval_kinetics.KINETICSclassification(
    ground_truth_filename=gt_json_file_path,
    prediction_filename=val_json_file_path,
    verbose=True,
    top_k=5
)
evaluator.evaluate()

>>> [INIT] Loaded annotations from validation subset.
>>> 	Number of ground truth instances: 19906
>>> 	Number of predictions: 133470
>>> [RESULTS] Performance on ActivityNet untrimmed video classification task.
>>> 	Error@5: 0.4760373756656284
```

Do you have any suggestion to reproduce your results?

**Comment by BarakBat** [comment]
So you got 0.63 on top1? that's what he got in the paper I think, the 70%  was with 64 frames here you used 16.
I was able to download only 84% from the training data ... :(
